packages/jsii-java-runtime/project/src/main/java/software/amazon/jsii/Util.java

```diff
@@ -45,18 +46,46 @@ static String readString(final InputStream is) {
      * @param methodName The name of the method.
      * @return True if the name looks like getXxx.
      */
-    static boolean isJavaPropertyMethod(final String methodName) {
-        return (methodName.length() > PROPERTY_METHOD_PREFIX_LEN
-                && (methodName.startsWith("get") || methodName.startsWith(("set"))));
+    static boolean isJavaPropertyMethod(final Method method) {
+        final String methodName = method.getName();
+        if (methodName.length() <= PROPERTY_METHOD_PREFIX_LEN) {
+            // Needs to have at least one character after the "get" or "set" prefix
+            return false;
+        }
+        if (methodName.startsWith("get") && method.getParameterCount() == 0) {
+            // Require something else than a lowercase letter after the "get" prefix
+            return !Character.isLowerCase(methodName.charAt(3));
+        }
+        if (methodName.startsWith("set") && method.getParameterCount() == 1) {
+            // Require something else than a lowercase letter after the "get" prefix
+            return !Character.isLowerCase(methodName.charAt(3))
+                // Require a matching getter
+                && isMatchingGetterPresent(method.getName().replaceFirst("set", "get"),
+                    method.getParameterTypes()[0],
+                    method.getDeclaringClass());
+        }
+        return false;
+    }
+
+    private static boolean isMatchingGetterPresent(final String getterName, final Class<?> returnType, final Class<?> declaring) {
+        try {
+            final Method getter = declaring.getDeclaredMethod(getterName);
+            return getter.getReturnType().equals(returnType);
+        } catch (final NoSuchMethodException nsme) {
+            return declaring.equals(Object.class)
+                                                  ? false
```

Comment:
This indent is weird...? Should be just indented one below the previous line.

Reply:
Uh - yeah!